### DISCUSSION TITLE
Using json?

### DISCUSSION THREAD

**Discussion by CredibleOpossum** [codefaux/persistence#13] [Ideas]
I was wondering how open you would be for replacing/changing the data_store file? In my own mod I simply use a lua library to [convert an object to a json string](https://github.com/CredibleOpossum/noita-research-mod/blob/main/init.lua#L138). It would probably be a little less efficient storage wise but would be easier and (I believe) improve forward compatibility.

I would be up to make a pull request if that's something you'd like.

**Comment by codefaux** [comment]
I appreciate the thought, and while I do like the appeal of serializing into json, I'm not interested in that change for this mod.

Two reasons; 
- aside from storage inside mod settings (which I'm using for spell loadouts) there is no persistant storage without requiring "allow unsafe  mods" and I am unwilling to require this. If there's an option I'm not aware of, I'm interested in hearing about it, but unwilling to convert to it for this mod.
- I am (presently) unwilling to make breaking changes, and don't want to need to maintain version-upgrade routines.

Largely my stance on this is not permanent. The mod is in a state of flux right now, where I intend to do some large refactors on existing code if I can find time and mental energy for them, and I still intend to add a few large features. Once I've got those features in place, I'll be -slightly- more open to things like refactoring, and possibly forward changes to the data store which continue to respect old versions.

**Reply by CredibleOpossum** [reply]
Am I misunderstanding something? What is the problem with mod settings? 

Yes, all good. I just believe the current implementation is pretty rigid and strange. If anyone attempts to contribute I'd imagine this file would be a barrier that most wouldn't bother with. JSON might not be the best solution but something that would allow for new variables to be added without a bunch of boilerplate needing to be changed would definitely be a good start.

I think lua has an inherit issue of being too loose with rules and control flow, making refactoring more important to any reasonably scaled project.
